cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -0,0 +1,309 @@
+/* file: bf_knn_impl.i */
+/*******************************************************************************
+* Copyright 2014-2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef __BF_KNN_IMPL_I__
+#define __BF_KNN_IMPL_I__
+
+#include "algorithms/engines/engine.h"
+#include "services/daal_defines.h"
+
+#include "algorithms/k_nearest_neighbors/bf_knn_classification_model.h"
+#include "src/algorithms/k_nearest_neighbors/bf_knn_classification_train_kernel.h"
+#include "src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_kernel.h"
+#include "src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_model_ucapi_impl.h"
+
+#include "data_management/data/numeric_table.h"
+#include "services/env_detect.h"
+#include "src/services/service_data_utils.h"
+#include "src/services/service_utils.h"
+#include "src/services/service_defines.h"
+#include "src/threading/threading.h"
+#include "src/data_management/service_numeric_table.h"
+
+#include "src/algorithms/service_kernel_math.h"
+#include "src/algorithms/service_sort.h"
+
+#include "src/externals/service_math.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace bf_knn_classification
+{
+namespace internal
+{
+const size_t classBufSize = 10;
```

Comment:
Consider better variable name and constexpr

Reply:
Switched to DEFINE and better name